.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,42 @@
+name: "PR Build"
+
+on:
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+        env:
+          GITHUB_TOKEN: ${{ secrets.TOKEN }}
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          server-id: github # Value of the distributionManagement/repository/id field of the pom.xml
+          settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+      - name: Check if submodule has been pulled
```

Comment:
not sure if this is needed. Won't the build fail if the submodule isn't pulled?

Reply:
no, it wont fail, just that opamp-proto jar will not be included in the final jar